### PR TITLE
Support no_std environment

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch supports embedded devices.
When the feature `std` is disabled, this crate only uses `core` and `alloc` crates, and disables `std` crate.

Currently, this branch uses `alloc` crate, but some environment does not support dynamic allocation, so I will add a feature to switch the allocation mechanism.

Due to this change, I remove random tests.

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -27,6 +31,7 @@ impl fmt::Display for DaachorseError {
     }
 }
 
+#[cfg(feature = "std")]
 impl Error for DaachorseError {}
```
Comment (kampersanda) [inline src/errors.rs:35]: It seems that `DaachorseError` implementations for `std` features in `errors` module are not needed.
Can we remove them?

**Review by kampersanda — COMMENTED** [review]
Thanks! Good modification!

File: src/errors.rs (hunk 1 above)
Comment (vbkaisetsu) [reply src/errors.rs:35]: Errors are often propagated as `Box<dyn std::errors::Error>`, so I think this is necessary.

Comment (kampersanda) [reply src/errors.rs:35]: I see.

**Review by kampersanda — APPROVED** [review]

**Pull request merged**